### PR TITLE
integration-tests/smoke/ccip: add max batch size test

### PR DESCRIPTION
**Pull request by makramkd**
Ticket: https://smartcontract-it.atlassian.net/browse/CCIP-4168

### CODE REVIEW

File: integration-tests/smoke/ccip/ccip_batching_test.go [hunk 1]
```diff
@@ -226,6 +229,77 @@ func Test_CCIPBatching(t *testing.T) {
...
+		var i = 0
+		for i < len(transactors) {
+			select {
+			case err := <-errs:
+				require.NoError(t, err)
+				i++
+			case <-ctx.Done():
+				require.FailNow(t, "didn't get all errors before test context was done")
+			}
+		}
```
Comment (dimkouv) [inline integration-tests/smoke/ccip/ccip_batching_test.go:289]: nit: maybe cleaner with `sync.WaitGroup`

**Review by dimkouv — APPROVED** [review]

File: integration-tests/smoke/ccip/ccip_batching_test.go (hunk 1 above)
Comment (makramkd) [reply integration-tests/smoke/ccip/ccip_batching_test.go:289]: Unfortunately, need the errors from the channel to check them

File: integration-tests/smoke/ccip/ccip_batching_test.go [hunk 2]
```diff
@@ -226,6 +229,77 @@ func Test_CCIPBatching(t *testing.T) {
...
+
+	t.Run("max evm batch size", func(t *testing.T) {
+		var (
+			sourceChain = sourceChain1
+			otherSender = mustNewTransactor(t, e.Env.Chains[sourceChain])
+			transactors = []*bind.TransactOpts{
+				e.Env.Chains[sourceChain].DeployerKey,
+				otherSender,
+			}
+			errs = make(chan error, len(transactors))
```
Comment (mateusz-sekara) [inline integration-tests/smoke/ccip/ccip_batching_test.go:247]: I guess channel's size should be 0 or 1 https://github.com/uber-go/guide/blob/master/style.md#channel-size-is-one-or-none

**Review by mateusz-sekara — APPROVED** [review]

**Review by 0xnogo — APPROVED** [review]

File: integration-tests/smoke/ccip/ccip_batching_test.go (hunk 2 above)
Comment (makramkd) [reply integration-tests/smoke/ccip/ccip_batching_test.go:247]: Actually did not work with an unbuffered channel, I tried that initially

**Pull request merged**